src/data_processing/wiki_to_netflix.py

```diff
@@ -0,0 +1,136 @@
+import requests
+import csv
+
+base_dir = os.path.join(os.path.dirname(__file__), '../data/')
+user_agent = 'Noisebridge MovieBot 0.0.1/Audiodude <audiodude@gmail.com>'
+
+missing_count = 0
+num_rows = 42
+
+# Reading netflix text file
+def read_netflix_txt(txt_file):
+    with open(txt_file, "r", encoding = "ISO-8859-1") as netflix_data:
+        netflix_list = [line.rstrip().split(',', 2) for line in netflix_data.readlines()[0:num_rows]]
+    return(netflix_list)
+
+# Writing netflix csv file
+def create_netflix_csv(csv_name, data_list):   
+    with open(csv_name, 'w') as netflix_csv:
+        csv.writer(netflix_csv).writerows(data_list)
+
+# Extracting movie info from Wiki data
+def wiki_feature_info(data, key):
+    if len(data['results']['bindings']) < 1 or key not in data['results']['bindings'][0]:
+        return 'NA'
+    if key == 'genreLabel':
+        return list({d['genreLabel']['value'] for d in data['results']['bindings'] if 'genreLabel' in d})
+    return data['results']['bindings'][0][key]['value'].split('/')[-1]
+
+# Adding movie info to netflix csv file
+def add_movie_info_to_csv(data_csv, movies, genres, directors):
+    with open(data_csv, 'r') as csv_file:
+        csv_reader = csv.reader(csv_file)
+        data_list = list(csv_reader)
+        
+    for i in range(0, len(data_list)-1):
+        if movies[i] or genres[i] or directors[i] != "NA":
+            data_list[i].extend((movies[i], genres[i], directors[i]))
+            
+    with open(data_csv, 'w') as csv_file:
+        csv_writer = csv.writer(csv_file)
+        csv_writer.writerows(data_list)  
+
+# Getting list of movie IDs, genre IDs, and director IDs from request
+def wiki_query(data_csv, user_agent):
+    wiki_movie_ids = []
+    wiki_genres = []
+    wiki_directors = []
+    
+    # csv_reader = list(csv.reader(open(data_csv)))
+    
+    for row in data_csv:
+        if row[1] != "NULL":
```

Comment:
Same thing as on the other PR, invert your ifs. Instead of an if, with 100 lines of code indented underneath it, reverse the if condition and exit early. So:

```
if row[1] == 'NULL':
  continue
```

And then outdent the rest of the stuff in the for loop.

Reply:
Fixed